### ISSUE TITLE
Template installation broken

### ISSUE THREAD

**Issue by bhcopeland** [elastic/ansible-elasticsearch#119]
```
TASK [ansible-elasticsearch : Get template files] ******************************
changed: [elk1]

TASK [ansible-elasticsearch : Install template(s)] *****************************
failed: [elk1] (item=logstash) => {"cmd": "curl -sL -XPUT http://localhost:9200/_template/logstash -d ''", "failed": true, "invocation": {"module_args": {"_raw_params": "curl -sL -XPUT http://localhost:9200/_template/logstash -d @/etc/elasticsearch/templates/logstash.json", "_uses_shell": false, "chdir": null, "creates": null, "executable": null, "removes": null, "warn": true}, "module_name": "command"}, "item": "logstash", "msg": "[Errno 2] No such file or directory", "rc": 2}
```

I had a look on the server:
```
ls /etc/elasticsearch/templates/
logstash
```
Doesn't seem to have renamed the file.

I tried renmaing file to logstash.json but still doesnt work.

```
failed: [elk1] (item=logstash) => {"cmd": "curl -sL -XPUT http://localhost:9200/_template/logstash -d ''", "failed": true, "invocation": {"module_args": {"_raw_params": "curl -sL -XPUT http://localhost:9200/_template/logstash -d @/etc/elasticsearch/templates/logstash.json", "_uses_shell": false, "chdir": null, "creates": null, "executable": null, "removes": null, "warn": true}, "module_name": "command"}, "item": "logstash", "msg": "[Errno 2] No such file or directory", "rc": 2}
```

However, if I run the above command manually it executes okay.

Above tests was done with ansible-2.0.2.0-2.

I upgraded to 2.1.0.0-1 and got another error message:

```
[DEPRECATION WARNING]: Skipping task due to undefined Error, in the future this will be a fatal error.: 'es_templates_fileglob' is undefined.

This feature will be removed in a future release. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
fatal: [elk1]: FAILED! => {"failed": true, "msg": "'item' is undefined"}
```

I fixed this by changing {{ es_templates_fileglob }} to templates/

```
- name: Copy templates to elasticsearch
  copy: src={{ item }} dest=/etc/elasticsearch/templates owner={{ es_user }} group={{ es_group }}
  with_fileglob: templates
```

But it still fails for me :(.

**Comment by bhcopeland** [comment]
Right, so it was because curl was not installed. So "No such file or directory" was just confusing.

Probably should make curl a dependency.

**Comment by gingerwizard** [comment]
@bhcopeland  or remove the dependency on curl all together!

thoughts @dliappis?

**Comment by bhcopeland** [comment]
@gingerwizard indeed. Probably should use the ansible [URI module](docs.ansible.com/uri_module.html).

```with_items: "{{ resultstemplate.stdout_lines }}"```

Breaks on latest stable Ansible.

Which also mean es_scripts_fileglob will also be broken.

**Comment by strootman** [comment]
The problem with the `URI module` is the `httplib2` dependency;

> The dependency on httplib2 was removed in Ansible 2.1

Which, from the docs, is no longer a problem in v2.1. Unless you state that only versions >= Ansible 2.1 are supported, I think you may have to wait on switching to the URI module (unless you're ok with the added complexity of managing the `httplib2` dependency as well).

**Comment by gingerwizard** [comment]
@strootman @bhcopeland the plan is to move to the lib module.  I use this for the new xpack+shield support which i hope to issue a pull request for shortly.  I actually have another issue with 2.1 - https://github.com/ansible/ansible/issues/15915

Currently im running all tests with 2.0.2 and stating urllib2 is required.  I may therefore delay the merge and state > 2.1.1 is required.

@bhcopeland can you be more specific re the issues with with_items? ticket?

**Comment by gingerwizard** [comment]
Will be addressed through
https://github.com/elastic/ansible-elasticsearch/issues/128
